src/components/Nav/GlobalNav/SearchDetail.js

```diff
@@ -0,0 +1,237 @@
+import React, { useState } from 'react';
+import { BsSearch } from 'react-icons/bs';
+import styled from 'styled-components/macro';
+import Location from './modals/Location';
+import Calendar from './modals/Calendar';
+import Guest from './modals/Guest';
+import variables from '../../../styles/variables';
+import theme from '../../../styles/theme';
+
+const SearchDetail = ({
+  setIsOpenSearchBar,
+  modalRef,
+  onChange,
+  endDate,
+  startDate,
+  increaseGuestNum,
+  decreaseGuestNum,
+  guestNum,
+  setSelectedLocation,
+  selectedLocation,
+  dateToString,
+}) => {
+  const [currentId, setCurrentId] = useState(0);
+  const [dateModalIsOpen, setDateModalIsOpen] = useState(false);
+  const [locationModalIsOpen, setLocationModalIsOpen] = useState(false);
+  const [guestModalIsOpen, setGuestModalIsOpen] = useState(false);
+
+  const clickHandler = id => {
+    setCurrentId(id);
+  };
+
+  const overLayClick = e => {
+    if (modalRef.current === e.target) {
+      setDateModalIsOpen(false);
+      setLocationModalIsOpen(false);
+      setGuestModalIsOpen(false);
+      setIsOpenSearchBar(false);
+      setCurrentId(0);
+    }
+  };
+
+  const ModalComponents = {
+    1: (
+      <Location
+        locationModalIsOpen={locationModalIsOpen}
+        currentId={currentId}
+        setSelectedLocation={setSelectedLocation}
+      />
+    ),
+    2: (
+      <Calendar
+        dateModalIsOpen={dateModalIsOpen}
+        currentId={currentId}
+        onChange={onChange}
+        startDate={startDate}
+        endDate={endDate}
+      />
+    ),
+    3: (
+      <Guest
+        guestModalIsOpen={guestModalIsOpen}
+        currentId={currentId}
+        increaseGuestNum={increaseGuestNum}
+        decreaseGuestNum={decreaseGuestNum}
+        guestNum={guestNum}
+      />
+    ),
```

Comment:
이렇게 바꾸는건 어때요? 1, 2, 3 이 어떤 컴포넌트를 의미하는지는 이 코드를 봐야 알수 있기 때문에 서제스쳔 남겨놓습니다. 

```suggestion
    location: (
      <Location
        locationModalIsOpen={locationModalIsOpen}
        currentId={currentId}
        setSelectedLocation={setSelectedLocation}
      />
    ),
    calendar: (
      <Calendar
        dateModalIsOpen={dateModalIsOpen}
        currentId={currentId}
        onChange={onChange}
        startDate={startDate}
        endDate={endDate}
      />
    ),
    guest: (
      <Guest
        guestModalIsOpen={guestModalIsOpen}
        currentId={currentId}
        increaseGuestNum={increaseGuestNum}
        decreaseGuestNum={decreaseGuestNum}
        guestNum={guestNum}
      />
    ),
```